components/electric/lib/electric/replication/postgres_connector_sup.ex

```diff
@@ -21,44 +23,39 @@ defmodule Electric.Replication.PostgresConnectorSup do
   @impl Supervisor
   def init(connector_config) do
     origin = Connectors.origin(connector_config)
-    name = name(origin)
-    Electric.reg(name)
 
-    postgres_producer = Postgres.LogicalReplicationProducer.name(origin)
-    postgres_producer_consumer = Postgres.MigrationConsumer.name(origin)
+    logical_replication_producer = Postgres.LogicalReplicationProducer.name(origin)
+    migration_consumer = Postgres.MigrationConsumer.name(origin)
+
     write_to_pg_mode = Connectors.write_to_pg_mode(connector_config)
 
     migration_consumer_opts = [
-      producer: postgres_producer,
+      producer: logical_replication_producer,
       refresh_subscription: write_to_pg_mode == :logical_replication
     ]
 
-    writer_config = [conn_config: connector_config, producer: SatelliteCollectorProducer.name()]
+    writer_module_opts = [
+      conn_config: connector_config,
+      producer: SatelliteCollectorProducer.name(origin)
+    ]
 
     children = [
+      {SchemaCache, connector_config},
+      {SatelliteCollectorProducer, connector_config},
+      {Postgres.LogicalReplicationProducer, connector_config},
       %{
-        id: :postgres_schema_cache,
-        start: {SchemaCache, :start_link, [connector_config]}
-      },
-      {SatelliteCollectorProducer,
-       name: SatelliteCollectorProducer.name(), write_to_pg_mode: write_to_pg_mode},
-      %{
-        id: :postgres_producer,
-        start: {Postgres.LogicalReplicationProducer, :start_link, [connector_config]}
-      },
-      %{
-        id: :postgres_migration_consumer,
+        id: Postgres.MigrationConsumer,
         start:
           {Postgres.MigrationConsumer, :start_link, [connector_config, migration_consumer_opts]}
       },
```

Comment:
Nitpick: if you're removing ids here, why not do this too?

```suggestion
{Postgres.MigrationConsumer, {connector_config, migration_consumer_opt}}
```

Reply:
sorry, the suggestion built up incorrectly, it's just over the migration consumer

Reply:
Perfect! I overlooked the tuple approach myself.